doc/system-ready/system-ready-HLD.md

```diff
@@ -0,0 +1,482 @@
+1. core services:
+    - swss, bgp, teamd, pmon, syncd, database, mgmt-framework are identified as core services and a list of these services are maintained predefined.
+
+2. all services:
+    - It covers the enabled services from FEATURE table of CONFIG_DB
+    - Also, since the idea is to cover only the sonic services but not the general system services, sysmonitor tracks services under "multi-user.target" and "sonic.target"
+    - This covers all the sonic docker services and most of the sonic host services.
+
+
+## 4.4 System ready Framework business logic
+
+The system ready framework design should just not only display the current status of the services in the system 
+but align the services within framework to flag the status as "not up" or "Failed" if the service is not running when it was intended to be running.
+
+- A 'core' service must be active and running and UP_STATUS marked by docker app should be True to be considered OK, otherwise flag it as Failed
+- For 'all' services:
+    - Loaded, enabled/enabled-runtime/static, active & running, active & exited state services are considered 'OK'.
+    - For active and running services, UP_STATUS marked by docker app should be True to be considered 'OK'.
+    - Failed state services are considered 'Failed'.
+    - Activating state services are considered as 'Starting'.
+    - Deactivating state services are considered as 'Stopping'.
+    - Inactive state services category:
+        - oneshot services are considered as 'OK'.
+        - Special services with condition pathexists check failure are considered as 'OK'.
+        - Other services in inactive state are considered to be 'Failed'.
+    - Any service type other than oneshot if by default goes to inactive state, RemainAfterExit=yes entry needs to be added to their service unit file to be inline with the framework.
+
+
+## 4.5 Provision for docker apps to mark closest UP status
+
+The feature provides framework for docker services to mark its closest UP status. 
+
+### 4.5.1 CONFIG_DB Changes (init_cfg.json)
+
+Docker apps marking their UP status in STATE_DB will input an entry in FEATURE table of CONFIG_DB with check_up_status flag set to True through /etc/sonic/init_cfg.json file change.
+Sysmonitor checks for the check_up_status flag in CONFIG_DB before reading the app ready status from STATE_DB. 
+If the flag does not exist or if set to False, then sysmonitor will not read the app ready status but just checks the running status of the service.
+
+
+```
+- Schema in /etc/sonic/init_cfg.json
+  This json file will be fed to FEATURE table of CONFIG_DB during bootup.
+  This json file will be factory default and no config command will be provided for "check_up_status" entry to be updated in CONFIG_DB later.
+           {
+              "FEATURE": {
+                 "<dockername>": {
+                     ...
+                     "state": "enabled",
+                     "check_up_status": "True"
```

Comment:
Feature table is for docker based features. Can we introduce a new table for host services like caclmgrd, hostcfgd?

Reply:
Agreed, Will introduce a new table in CONFIG_DB called HOST_FEATURE for this purpose.